src/foam/u2/view/PasswordView.js

```diff
@@ -18,7 +18,85 @@
 foam.CLASS({
   package: 'foam.u2.view',
   name: 'PasswordView',
-  extends: 'foam.u2.tag.Input',
+  extends: 'foam.u2.View',
 
-  properties: [ [ 'type', 'password' ] ]
+  css: `
+    .input-field-container {
+      position: relative;
+    }
+    .input-image {
+      position: absolute;
+      width: 24px;
+      height: 24px;
+      bottom: 8px;
+      right: 6px;
+    }
+    .input-field {
+      padding-right: 30px;
+    }
+  `,
+
+  properties: [
+    'data',
+    'type',
+    {
+      name: 'visibilityIcon',
+      value: '/foam2/src/foam/u2/images/visibility.svg'
+    },
+    {
+      class: 'Boolean',
+      name: 'passwordInvisible',
+      value: true
+    },
+    {
+      class: 'Boolean',
+      name: 'passwordIcon',
+      defaultValue: false
+    },
+    'inputElement'
+  ],
+
+  methods: [
+    function initE() {
+      this.SUPER();
+
+      this.addClass(this.myClass()).start().
+        addClass('input-field-container').
+        start('input', {
+          type: this.type
+        }, this.inputElement$).
+          addClass('input-field').addClass('full-width-input-password').
+        end().
+        start('img').show(this.passwordIcon$).addClass('input-image').
+          attr('src', this.visibilityIcon$).on('click', this.visible).
+        end().
+      end();
+
+      this.inputElement.attrSlot(
```

Comment:
You don't need to do this. the start('input') is mapped to a foam.u2.tag.Input element which already supports this logic. Just set it's onKey property and then bind to its data$.